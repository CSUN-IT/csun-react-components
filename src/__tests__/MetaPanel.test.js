import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import MetaPanel from '../../src/components/meta/Panel';

configure({ adapter: new Adapter() });

describe('MetaPanel', () => {
  it('Should be defined', () => {
    expect(MetaPanel).toBeDefined();
  });

  it('Renders component correctly', () => {
    const tree = shallow(
      <MetaPanel
        panelHeader={{ text: 'Panel' }}
        panelContent={{ text: 'This is a panel w/ a header' }}
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Renders children correctly', () => {
    const tree = mount(
      <MetaPanel
        panelHeader={{ text: 'Panel' }}
        panelContent={{ text: 'This is a panel w/ a header' }}
      />,
    );
    expect(tree.find(MetaPanel)).toHaveLength(1);
  });
});
