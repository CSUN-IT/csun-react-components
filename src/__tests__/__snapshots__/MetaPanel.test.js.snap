// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaPanel Renders component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MetaPanel
    className=""
    panelContent={
      Object {
        "text": "This is a panel w/ a header",
      }
    }
    panelFooter={null}
    panelHeader={
      Object {
        "text": "Panel",
      }
    }
    panelImage={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="panel__header"
        >
          <strong>
            Panel
          </strong>
        </div>,
        null,
        <div
          className="panel__content"
        >
          This is a panel w/ a header
        </div>,
        null,
      ],
      "className": " panel",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <strong>
            Panel
          </strong>,
          "className": "panel__header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Panel",
          },
          "ref": null,
          "rendered": "Panel",
          "type": "strong",
        },
        "type": "div",
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This is a panel w/ a header",
          "className": "panel__content",
        },
        "ref": null,
        "rendered": "This is a panel w/ a header",
        "type": "div",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="panel__header"
          >
            <strong>
              Panel
            </strong>
          </div>,
          null,
          <div
            className="panel__content"
          >
            This is a panel w/ a header
          </div>,
          null,
        ],
        "className": " panel",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <strong>
              Panel
            </strong>,
            "className": "panel__header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Panel",
            },
            "ref": null,
            "rendered": "Panel",
            "type": "strong",
          },
          "type": "div",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This is a panel w/ a header",
            "className": "panel__content",
          },
          "ref": null,
          "rendered": "This is a panel w/ a header",
          "type": "div",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
